src/creep.prototype.js

```diff
@@ -573,6 +586,122 @@ Creep.prototype.routeTo = function routeCreep(target, opts) {
   });
 };
 
+function getTargetAndRange(creep) {
+  let range = 1;
+  let target;
+  if (creep.memory.target) {
+    if (_.isNumber(creep.memory.target)) {
+      target = Game.getObjectById(creep.memory.target)
+    } else if (Array.isArray(creep.memory.target)) {
+      target = new RoomPosition(...creep.memory.target);
+    }
+    if (_.isNumber(creep.memory.range)) {
+      range = creep.memory.range;
+    }
+  }
+  return [target, range];
+}
+
+function moveCreepOutOfTheWay(creep, target, range, creepsToMove, nextPositions, creepMoves) {
+  const availableSpots = [...walkBox(creep.pos)].filter(([x, y]) => {
+    const newPos = new RoomPosition(x, y, creep.room.name);
+    if (target && newPos.getRangeTo(target) > range) {
+      return false;
+    }
+    for (let curr of newPos.look()) {
+      if (
+        (curr.type === 'terrain' && curr.terrain === 'wall')
+        || (
+          curr.type === 'structure'
+          && !curr.structure instanceof StructureRoad
+          && !curr.structure instanceof StructureContainer
+          && !(curr.structure instanceof StructureRampart && (curr.my || (!curr.my && curr.isPublic)))
+        )
+      ) {
+        return false;
+      }
+    }
+    const creepsAtMySpot = newPos.lookFor(LOOK_CREEPS);
+    if (creepsAtMySpot.length) {
+      const creepAtMySpot = creepsAtMySpot[0];
+      if (!creepAtMySpot.my) {
+        return false;
+      } else if (creepAtMySpot.my && !creepsToMove[creepAtMySpot.name]) {
+        // Try to move this creep out of the way....
+        const [target, range] = getTargetAndRange(creepAtMySpot);
+        creepsToMove[creepAtMySpot.name] = { target, range }
+        moveCreepOutOfTheWay(creepAtMySpot, target, range, creepsToMove, nextPositions, creepMoves);
+      }
+    }
+    if (
+      !nextPositions.find(pos => pos.x === newPos.x && pos.y === newPos.y)
+      && !creepMoves.find(([creep, pos]) => pos.x === newPos.x && pos.y === newPos.y)
+    ) {
+      return true;
+    }
+  });
+  if (target) {
+    let newPos;
+    const keepRangeSpot = availableSpots.find(([x, y]) => {
+      newPos = new RoomPosition(x, y, creep.room.name);
+      const newRange = newPos.getRangeTo(target);
+      if (
+         newRange === range
+        || (newRange <= range && newRange > creep.pos.getRangeTo(target))
+      ) {
+        return true;
```

Comment:
Prefer a position that maintains or increases range (because I think it looks cooler)